app/api/init_api.py

```diff
@@ -64,6 +67,57 @@ def openapi(self) -> dict[str, Any]:
         return self.openapi_schema
 
 
+@asynccontextmanager
+async def lifespan(asgi_app: BanchoAPI) -> AsyncIterator[Never]:
+    app.state.loop = asyncio.get_running_loop()
```

Comment:
potentially could get rid of state in the future and use the lifespan instead. worth discussing at the very least. left alone for now